### PR TITLE
Split exec/capture

### PR DESCRIPTION
**Pull request by jleeothon**

### CODE REVIEW

File: lib/eksekuter.rb [hunk 1]
```diff
@@ -14,71 +14,76 @@ def initialize logger: nil
...
     pid = spawn(params.fetch(:env), *params.fetch(:cmd), params.fetch(:opts))
-    close_streams streams
     _, process_status = Process.wait2(pid)
     process_status
   end
 
-  def close_streams streams
-    out_writable = streams.fetch(:out).fetch(:writable)
-    err_writable = streams.fetch(:err).fetch(:writable)
-    out_writable&.close if out_writable != STDOUT
```
Comment (jleeothon) [inline lib/eksekuter.rb]: On a second thought, I see a problem with this piece of code:

- These streams could have been created by ourselves in `get_out_err_streams`. Closing them is fine.
- If the streams were provided by the user, it's awkward to decide to close them without consulting the user.
- How do we distinguish between streams provided by the user and streams created by us? We have to add moar code?

File: lib/eksekuter.rb [hunk 2]
```diff
@@ -14,71 +14,60 @@ def initialize logger: nil
...
-    env, cmd = separate_env_and_cmd(args)
-    # p opts
-    capture = opts.delete(:capture) || false
-    streams = get_out_err_streams(capture)
+  alias kernel_spawn spawn
 
-    set_streams_in_opts streams, opts
-
-    params = { env: env, cmd: cmd, opts: opts, block: block }
+  # Wraps around Kernel#spawn so that the return value is a EksekResult.
```
Comment (tacconthommy) [inline lib/eksekuter.rb]: *an* EksekResult 🙈

File: lib/eksekuter.rb [hunk 3]
```diff
@@ -14,71 +14,60 @@ def initialize logger: nil
...
+    params = { env: env, cmd: cmd, opts: opts }
+    process_status = spawn_and_get_status(params)
+    assemble_result(params, process_status)
+  end
 
-    process_status = run_process(params, streams)
-    assemble_result(params, process_status, streams)
+  # Like Eksekuter#spawn but the :out and :err options are ignored; instead
+  # the output is attached to the EksekResult object returned.
+  # Returns a EksekResult object which getters :stdout and :stderr containing
```
Comment (tacconthommy) [inline lib/eksekuter.rb]: *an* EksekResult object

File: lib/eksekuter.rb (hunk 1 above)
Comment (tacconthommy) [reply lib/eksekuter.rb]: Is this still relevant? The code you are talking about has been removed...

File: lib/eksekuter.rb [hunk 4]
```diff
@@ -14,71 +14,60 @@ def initialize logger: nil
...
+    err_write.close
+    assemble_result(params, process_status, out_read, err_read)
   end
 
   private
 
-  def separate_env_and_cmd args
-    env = args[0].is_a?(Hash) ? args.shift : {}
+  def split_env_and_cmd args
+    args2 = args.clone
```
Comment (tacconthommy) [inline lib/eksekuter.rb]: Can we not name this `args2` but maybe `cmd_args` or something?

Comment (jleeothon) [reply lib/eksekuter.rb]: Done :)

File: lib/eksekuter.rb (hunk 1 above)
Comment (jleeothon) [reply lib/eksekuter.rb]: Nope!

But it helps explain why I moved the code around in a way that this function is not useful anymore

**Pull request merged**